DataFormats/TrackerCommon/interface/TrackerTopology.h

```diff
@@ -576,7 +577,33 @@ class TrackerTopology {
   std::string print(DetId detid) const;
 
   SiStripDetId::ModuleGeometry moduleGeometry(const DetId &id) const; 
+  
+  // Represents a bitmask that is only valid in some subdetector. 
+  // However, checking validity still has to be done by the caller.
+  struct BitMask {
+    uint32_t startBit_;
+    uint32_t mask_;
+    int32_t  subdet;
+    uint32_t apply(DetId id) const { return ((id.rawId()>>startBit_) & mask_); };
+    bool     valid(DetId id) const { return id.subdetId() == subdet; };
+  };
+
+  std::map<std::string, BitMask> namedPartitions() const { 
+    std::map<std::string, BitMask> namedPartitions_{
```

Comment:
On 4/18/16 9:14 AM, Marcel Schneider wrote:
> Regarding the constructor, things get more delicate with the
> initialization order then, and moving it to a default-initializer in the
> .h caused sth. that looks very much like a compiler bug. Also, it is
> returned by value anyways, so copying vs. in-place construction should
> not make a big difference.

it will not/should not be returned by value, if the namedPartitions_ is
available as a data member computed at costruction.